app/models/swipe.rb

```diff
@@ -0,0 +1,24 @@
+class Swipe < ActiveRecord::Base
+  belongs_to :swiper, foreign_key: :swiper_id, class_name: "User"
+  belongs_to :swipee, foreign_key: :swipee_id, class_name: "User"
+  belongs_to :restaurant
+
+  validate :cannot_swipe_on_self, on: :create
+
+  validates :swiper_id, uniqueness: {scope: [:swipee_id, :restaurant_id]}
+
+  def search_for_match(swipe_creator)
```

Comment:
This could maybe be better named match_exists? - it's returning a boolean as to whether the record exists. The current name suggests it returns the match.